internal/controller/managedzone_controller.go

```diff
@@ -105,6 +109,22 @@ func (r *ManagedZoneReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		}
 	}
 
+	// Ensure provider secret exists before continuing
+	err = r.Get(ctx, client.ObjectKey{Namespace: managedZone.Namespace, Name: managedZone.Spec.SecretRef.Name}, &v1.Secret{}, &client.GetOptions{})
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			secretMissing.WithLabelValues(managedZone.Name, managedZone.Namespace, managedZone.Spec.SecretRef.Name).Set(1)
```

Comment:
I know metrics was already in here, but it would probably be better in it's own "metrics" package as it really wasn't obvious what this was. One for later maybe as we add more metrics.